src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs

```diff
@@ -34,14 +34,21 @@ public void EmptyAsPath_ThrowsArgumentException()
         }
 
         [Theory, MemberData(nameof(PathsWithInvalidCharacters))]
-        public void PathWithInvalidCharactersAsPath_ThrowsArgumentException(string invalidPath)
+        [SkipOnTargetFramework(~TargetFrameworkMonikers.NetFramework)]
+        public void PathWithInvalidCharactersAsPath_Desktop(string invalidPath)
         {
-            if (invalidPath.Equals(@"\\?\") && !PathFeatures.IsUsingLegacyPathNormalization())
-                AssertExtensions.ThrowsAny<IOException, UnauthorizedAccessException>(() => Create(invalidPath));
-            else if (invalidPath.Contains(@"\\?\") && !PathFeatures.IsUsingLegacyPathNormalization())
-                Assert.Throws<DirectoryNotFoundException>(() => Create(invalidPath));
-            else
+            Assert.Throws<ArgumentException>(() => Create(invalidPath));
+        }
+
+        [ActiveIssue(27269)]
+        [Theory, MemberData(nameof(PathsWithInvalidCharacters))]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
+        public void PathWithInvalidCharactersAsPath_Core(string invalidPath)
+        {
+            if (invalidPath.Contains('\0'))
                 Assert.Throws<ArgumentException>(() => Create(invalidPath));
```

Comment:
Is there a param name we can include/test here?

Reply:
Not for desktop, but now that I've split it up I can (and will) add it for core.